presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -92,101 +92,223 @@ public class PrestoNativeQueryRunnerUtils
+        {
+            this.addStorageFormatToPath = addStorageFormatToPath;
+            return this;
+        }
 
-        if ("legacy".equals(security)) {
-            hivePropertiesBuilder.put("hive.allow-drop-table", "true");
+        public Builder setCacheMaxSize(Integer cacheMaxSize)
+        {
+            this.cacheMaxSize = cacheMaxSize;
+            return this;
+        }
+
+        public Builder setExtraProperties(Map<String, String> extraProperties)
+        {
+            this.extraProperties = extraProperties;
+            return this;
+        }
+
+        public Builder setSecurity(String security)
+        {
+            this.security = security;
+            return this;
+        }
+
+        public QueryRunner buildNativeHiveQueryRunner()
+                throws Exception
+        {
+            // The property "hive.allow-drop-table" needs to be set to true because security is always "legacy" in NativeQueryRunner.
+            ImmutableMap<String, String> hiveProperties = ImmutableMap.<String, String>builder()
+                    .putAll(getNativeWorkerHiveProperties(storageFormat))
+                    .put("hive.allow-drop-table", "true")
+                    .build();
+
+            ImmutableMap.Builder<String, String> coordinatorProperties = ImmutableMap.builder();
+            coordinatorProperties.put("native-execution-enabled", "true");
+            if (singleNodeExecutionEnabled) {
+                coordinatorProperties.put("single-node-execution-enabled", "true");
+            }
+
+            // Make query runner with external workers for tests
+            return HiveQueryRunner.createQueryRunner(
+                    ImmutableList.of(),
+                    ImmutableList.of(),
+                    ImmutableMap.<String, String>builder()
+                            .put("http-server.http.port", "8081")
+                            .put("experimental.internal-communication.thrift-transport-enabled", String.valueOf(useThrift))
+                            .putAll(getNativeWorkerSystemProperties())
+                            .putAll(coordinatorSidecarEnabled ? getNativeSidecarProperties() : ImmutableMap.of())
+                            .putAll(extraProperties)
+                            .build(),
+                    coordinatorProperties.build(),
+                    "legacy",
+                    hiveProperties,
+                    Optional.ofNullable(workerCount),
+                    Optional.of(Paths.get(addStorageFormatToPath ? dataDirectory.toString() + "/" + storageFormat : dataDirectory.toString())),
+                    getExternalWorkerLauncher("hive", serverBinary, cacheMaxSize, remoteFunctionServerUds, failOnNestedLoopJoin,
+                            coordinatorSidecarEnabled, enableRuntimeMetricsCollection, enableSsdCache),
+                    getNativeWorkerTpcdsProperties());
+        }
+
+        public QueryRunner buildJavaHiveQueryRunner()
```

Comment:
Rather than have this `buildXXXQueryRunner` pattern with different methods, I feel like it is more true to the builder pattern to instantiate a builder with a set of values that set the parameters for this, and then have a singular `build` method. Essentially the code becomes

```java
javaQueryRunnerBuilder()
    .setABC()
    .setXYZ()
    .build()
...

nativeQueryRunnerBuilder()
    .setABC()
    .setXYZ()
    .build()
```

In this way, you can still override some of the parameters. In the current approach it would be impossible to, say, remove the `hive.pushdown-filter-enabled` configuration. WDYT?

Reply:
+1. Agree with Zac.